taker/src/main.rs

```diff
@@ -169,6 +184,32 @@ impl Network {
 async fn main() -> Result<()> {
     let opts = Opts::parse();
 
+    let network = match opts.network {
+        Some(network) => network,
+        None => Network::Mainnet {
+            electrum: MAINNET_ELECTRUM.to_string(),
+            withdraw: None,
+        },
+    };
+
+    let maker = match opts.maker {
+        Some(maker) => maker,
+        None => match network {
+            Network::Mainnet { .. } => MAINNET_MAKER.to_string(),
+            Network::Testnet { .. } => TESTNET_MAKER.to_string(),
+            Network::Signet { .. } => bail!("No maker default URL configured for signet"),
+        },
+    };
+
+    let maker_id = match opts.maker_id {
+        Some(maker) => maker,
+        None => match network {
+            Network::Mainnet { .. } => parse_x25519_pubkey(MAINNET_MAKER_ID)?,
+            Network::Testnet { .. } => parse_x25519_pubkey(TESTNET_MAKER_ID)?,
+            Network::Signet { .. } => bail!("No maker default URL configured for signet"),
+        },
+    };
```

Comment:
All of this defaulting logic could be inside an `impl Opts {}` which would reduce the stuff `main` has to deal with.